lnrpc/routerrpc/router.proto

```diff
@@ -465,15 +409,15 @@ service Router {
     /**
     SendPayment attempts to route a payment described by the passed
     PaymentRequest to the final destination. The call returns a stream of
-    payment status updates.
+    payment updates.
     */
-    rpc SendPayment (SendPaymentRequest) returns (stream PaymentStatus);
+    rpc SendPayment (SendPaymentRequest) returns (stream lnrpc.Payment);
```

Comment:
Yes. Previously there wasn't really an incremental update other than the payment status code. No in-flight htlcs were ever streamed back, only the final set.

I thought about sending back a delta containing the changed fields only, but it may get a little complicated to use. The caller needs to correctly apply those deltas to get the full payment state. In the end, it is an optimization for which I think it isn't the time yet.

Reply:
Can totally changing response messages be avoided?

Reply:
It can be avoided, but because we are starting the move to mainlining `routerrpc` now that would be a missed chance to make the api consistent.

Reply:
Sure, I wish that there were a better upgrade path but it seems extremely disruptive to me and potentially there are less disruptive ways to go about it that still allow for a cleaner future API direction

Reply:
I feel it could be useful to wrap the return value in one of our standard return types (`SendPaymentResponse`). This would give use more flexibility going forward adding more data to it, independent of `TrackPayment`. 

`TrackPayment` would similarly have a `TrackPaymentResponse` that for now just wraps the `lnrpc.Payment`. Is there any reason at all that the two calls must have the same response type?

Reply:
`TrackPayment` is basically a streaming version of `ListPayments`. I can even see a future where both are integrated. In `ListPayments` we also return a list of plain `Payment` messages. `SendPayment` is a convenience method that combines initiating the payment and following up with `TrackPayment`.

I am wary of this kind of future proofing, but not strongly opposed to it. If another reviewer agrees that it is an improvement, I can add it. @cfromknecht @Roasbeef